### PR TITLE
Issue 19

### PR DESCRIPTION
**Pull request by BunnyJojo**
Added in Functions for all social media. You can add it from editing profile and update if needed.

### CODE REVIEW

**Comment by arna0008** [comment]
Looks good to me! I can add all my social media links, and embed a youtube video link.
Only issue is that the icons are sitting on top of the embedded video - we want them to move under.
screen shot 2018-03-12 at 6 43 19 pm

**Comment by amclean1980** [comment]
you've got to rebase on master before committing this branch to bring it up to speed.  Also, if you want to make 2 minor edits to allow viewing existing data when you edit

Update your GuideProfileController.php:

  public function editGuide () {
       $user = Auth::user();
       $guide = $user->guide;
       $certs = $guide->certifications->all();
       $video = $guide->video()->first();

       return view(‘editprofile’, [‘guide’ => $guide, ‘certs’ => $certs,
           ‘firstname’ => $user->firstname,
           ‘email’ => $user->email, ‘video’ => $video]);
   }

And editprofile.blade.php

          <div>
               {!! Form::open(array(‘route’ => ‘videos.store’)) !!}
               {{Form::label(‘YoutubeL’,‘YouTube URL :’)}}
               {{Form::text(‘Youtube_URL’, empty($video) ? null : $video->Youtube_URL, array(‘class’ => ‘form-control’, ‘required’ => ‘required’))}}
               {{Form::label(‘Facebook’,‘Facebook Link:’)}}
               {{Form::text(‘Facebook_URL’, empty($video) ? null : $video->Facebook_URL, array(‘class’ => ‘form-control’,‘required’ => ‘required’))}}
               {{Form::label(‘Twitter’,‘Twitter Link:’)}}
               {{Form::text(‘Twitter_URL’,empty($video) ? null : $video->Twitter_URL,array(‘class’ => ‘form-control’,‘required’ => ‘required’))}}
               {{Form::label(‘Instagram’,‘Instagram Link:’)}}
               {{Form::text(‘Instagram_URL’, empty($video) ? null : $video->Instagram_URL, array(‘class’ => ‘form-control’,‘required’ => ‘required’))}}
               {{Form::submit(‘Update’)}}
               {!! Form::close() !!}
           </div> (edited)
basically, get a reference to the video you want in the editGuide function of the controller and pass that reference to the blade file as video.  The in in the blade file, you can access variable video, and subsequently all the attribues.  Now, you instead of ‘null’ for the preset value, pass in the existing values. (edited)

File: AdventureInnovation/AdventureInnovation/app/Http/Controllers/GuideProfileController.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        return view('profile', ['guide' => $guide, 'certs' => $certs,
+            'firstname' => $user->firstname,
+            'email' => $user->email]);
+    }
+
+    public function editGuide () {
+        $user = Auth::user();
+        $guide = $user->guide;
+        $certs = $guide->certifications->all();
```
Comment (amclean1980) [inline AdventureInnovation/AdventureInnovation/app/Http/Controllers/GuideProfileController.php]: add a reference to the video, then pass that to the blade

File: AdventureInnovation/AdventureInnovation/resources/views/editprofile.blade.php [hunk 2]
```diff
@@ -178,15 +178,35 @@ function( response, status, xhr ) {
...
                     <i class="fa fa-facebook-official" style="padding:20px" aria-hidden="true"></i>
                 </div>
+                SOCIAL MEDIA BUTTONS -->
 
             </div>
+        <div class="panel panel-heading"><h4><i class="fa fa-paper-plane-o" aria-hidden="true"></i> Social Media</h4>
+        </div>
+            <div>
+                {!! Form::open(array('route' => 'videos.store')) !!}
+                {{Form::label('YoutubeL','YouTube URL :')}}
```
Comment (amclean1980) [inline AdventureInnovation/AdventureInnovation/resources/views/editprofile.blade.php]: update 'null' values to add in existing values if they exist

empty($video) ? null : $video->Youtube_URL

or something like that

File: AdventureInnovation/AdventureInnovation/database/seeds/VideosTableSeeder.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class VideosTableSeeder extends Seeder
+{
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        DB::table('videos')->insert([
```
Comment (amclean1980) [inline AdventureInnovation/AdventureInnovation/database/seeds/VideosTableSeeder.php]: do this

   public function run()
    {
        $user = User::where('email', '=', 'brennan@brennan.com')->firstOrFail();

        $video = new Video();
        $video->Youtube_URL = 'https://www.youtube.com/embed/tgbNymZ7vqY';
        $video->Facebook_URL = 'https://www.facebook.com/canada150th/';
        $video->Twitter_URL = 'https://twitter.com/canada150th?lang=en';
        $video->Instagram_URL = 'https://www.instagram.com/explore/tags/canada150/';
        $video->guide_id = $user->guide->id;

        $video->save();

    }

Comment (amclean1980) [reply AdventureInnovation/AdventureInnovation/database/seeds/VideosTableSeeder.php]: Sorry, wasn't clear.  This just ensures that  we're seeding a specific usesr with those videos

Comment (amclean1980) [reply AdventureInnovation/AdventureInnovation/database/seeds/VideosTableSeeder.php]: Also, the supplied link doesn't seem to embed properly ( ie, just shows up as text when i clear everything out and repopulate the DB).  If i change it to https://youtu.be/tgbNymZ7vqY, it seems to work
